File: src/main/java/voidchess/player/ki/StaticEvaluationInterface.java
package voidchess.player.ki;

import voidchess.board.ChessGameInterface;
import voidchess.player.ki.evaluation.Evaluated;

/**
 * @author stephan
 */
public interface StaticEvaluationInterface {
    Evaluated getPrimaryEvaluation(ChessGameInterface game, boolean forWhite);

    void addSecondaryEvaluation(ChessGameInterface game, boolean forWhite, Evaluated withPrimaryEvaluation);
}

File: src/main/kotlin/voidchess/player/ki/StaticEvaluationInterface.kt
package voidchess.player.ki

import voidchess.board.ChessGameInterface
import voidchess.player.ki.evaluation.Evaluated
import voidchess.player.ki.evaluation.Ongoing


interface StaticEvaluationInterface {
    fun getPrimaryEvaluation(game: ChessGameInterface, forWhite: Boolean): Ongoing
    fun addSecondaryEvaluation(game: ChessGameInterface, forWhite: Boolean, withPrimaryEvaluation: Evaluated)
}